assets/js/blocks/product-category/block.js

```diff
@@ -161,33 +208,39 @@ class ProductByCategoryBlock extends Component {
 				) }
 				<div className="wc-block-products-category__selection">
 					<ProductCategoryControl
-						selected={ attributes.categories }
+						selected={ currentAttributes.categories }
 						onChange={ ( value = [] ) => {
 							const ids = value.map( ( { id } ) => id );
-							setAttributes( { categories: ids } );
+							this.setAttributes( { categories: ids } );
 						} }
-						operator={ attributes.catOperator }
+						operator={ currentAttributes.catOperator }
 						onOperatorChange={ ( value = 'any' ) =>
-							setAttributes( { catOperator: value } )
+							this.setAttributes( { catOperator: value } )
 						}
 					/>
 					<Button isDefault onClick={ onDone }>
 						{ __( 'Done', 'woo-gutenberg-products-block' ) }
 					</Button>
+					<Button
+						className="wc-block-products-category__cancel-button"
+						title={ __( 'Cancel', 'woo-gutenberg-products-block' ) }
+						isLink
```

Comment:
How about [`isTertiary` for text styled button instead?](https://github.com/wordpress/gutenberg/tree/master/packages/components/src/button#props) This link-styled button has an awkward focus state:

<img width="433" alt="Screen Shot 2019-05-22 at 3 05 28 PM" src="https://user-images.githubusercontent.com/541093/58201569-635c0800-7ca3-11e9-8f8b-bc0ed77b7815.png">